ENH: Describe execution of MRIQC and how to add the derivatives (#292)
TheAxonLab/hcph-sops: docs/data-management/preliminary.md

            --url https://github.com/{{ secrets.data.gh_repo | default('<organization>/<repo_name>') }}.git \
            --publish-depends ria-storage
    ```
    

> - [ ] Create a sub-dataset to host the *MRIQC* derivatives. If you will not run *MRIQC* version 23.1.0, replace the version with the one you intend to run in the command below.

Reviewer: ```suggestion
- [ ] Create a sub-dataset to host the *MRIQC* derivatives.
    Remember to set the correct version of the container (in our case {{ settings.versions.mriqc }}).
```

Reviewer: Remember to add the config for `{{ settings.versions.mriqc }}` in `/study-settings.yml` so that the replacement happens consistently throughout the SOPs.

Add something like:

```yaml
versions:
  mriqc: '23.1.0'
  fmriprep: 'xxxx'
  dmriprep: 'xxxx'
```

    ``` shell
    cd /data/datasets/hcph-dataset

>     datalad create -d . derivatives/mriqc-23.1.0

Reviewer: ```suggestion
    datalad create -d . derivatives/mriqc-{{ settings.versions.mriqc }}
```

            Therefore, the execution of resource-intensive tasks such as *fMRIPrep* or building containers on login nodes can negatively impact the overall performance and responsiveness of the system for all users.
            Interactive sessions are a great alternative when available and **should** be used when creating the *DataLad* dataset.
            For example, in the case of systems operating SLURM, the following command would open a new interactive session:
            ```
            srun --nodes=1 --ntasks-per-node=1 --time=01:00:00 --pty bash -i
            ```

    - [ ] Install *DataLad*.
        Generally, the most convenient and user-sandboxed installation (i.e., without requiring elevated permissions) can be achieved by using *Conda*, but other alternatives (such as *lmod*) can be equally valid:

        === "Install *DataLad* with *Conda*"

            - [ ] Get and install *Conda* if it is not already deployed in the system:

                ``` shell
                wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
                bash Miniconda3-latest-Linux-x86_64.sh
                ```

            - [ ] Install *DataLad*:

> 
>                 ``` shell
>                 conda install -c conda-forge -y "datalad>=0.19" datalad-container
>                 ```
> 
>         === "Install *DataLad* in HPC with *lmod* enabled"
> 
>             - [ ] Check the availability and dependencies for a specific Python version (here we check 3.8.2):
> 
>                 ``` bash
>                 module spider Python/3.8.2
>                 ```
> 
>             - [ ] Load Python (please note `ml` below is a shorthand for `module load`)
> 
>                 ``` bash
>                 ml GCCcore/9.3.0 Python/3.8.2
>                 ```
> 
>             - [ ] Update *pip*:
> 
>                 ``` bash
>                 python -m pip --user -U pip
>                 ```
> 
>             - [ ] Install *DataLad*:
> 
>                 ``` bash
>                 python -m pip install --user "datalad>=0.19" datalad-container
>                 ```
> 
>     - [ ] Check datalad is properly installed, for instance:
> 
>         ``` shell
>         $ datalad --version
>         datalad 0.19.2
>         ```
> 
>         ??? bug "*DataLad* crashes (*Conda* installations)"
> 
>             *DataLad* may fail with the following error:
>             ``` py
>             ImportError: cannot import name 'getargspec' from 'inspect' (/home/users/cprovins/miniconda3/lib/python3.11/inspect.py)
>             ```
> 
>             In such a scenario, create a *Conda* environment with a lower version of Python, and re-install datalad
>             ``` shell
>             conda create -n "datalad" python=3.10
>             conda activate datalad
>             conda install -c conda-forge datalad datalad-container
>             ```
> 
>     - [ ] Configure your Git identity settings.
> 
>         ``` shell
>         cd ~
>         git config --global --add user.name "Jane Doe"
>         git config --global --add user.email doe@example.com
>         ```
> 

Reviewer: I'd prefer to keep this at the bottom where it was originally. Also, this change is unrelated to the PR.

```suggestion
```

Reviewer: Okay, I see the point now - there's a similar admonition for workstation installation. Let's address this somewhere else.

Reviewer: Actually, I don't see the point -- but still think we should defer this to a separate conversation.

            ```

        - [ ] Load Python (please note `ml` below is a shorthand for `module load`)

            ``` bash
            ml GCCcore/9.3.0 Python/3.8.2
            ```

        - [ ] Update *pip*:

            ``` bash
            python -m pip --user -U pip
            ```

        - [ ] Install *DataLad*:

            ``` bash
            python -m pip install --user "datalad>=0.19" datalad-container
            ```

- [ ] Check datalad is properly installed, for instance:

    ``` shell
    $ datalad --version
    datalad 0.19.2
    ```

    ??? bug "*DataLad* crashes (*Conda* installations)"

        *DataLad* may fail with the following error:
        ``` py
        ImportError: cannot import name 'getargspec' from 'inspect' (/home/users/cprovins/miniconda3/lib/python3.11/inspect.py)
        ```

        In such a scenario, create a *Conda* environment with a lower version of Python, and re-install datalad
        ``` shell
        conda create -n "datalad" python=3.10
        conda activate datalad
        conda install -c conda-forge datalad datalad-container
        ```

- [ ] Configure your Git identity settings.

    ``` shell
    cd ~

>     git config --global --add user.name "Jane Doe"
>     git config --global --add user.email doe@example.com
>     ```
> 

Reviewer: ```suggestion
### HPC users - instructions to install *DataLad*
When HPC is planned for processing, *DataLad* will be required on that system(s).

- [ ] Start an interactive session on the HPC cluster

    ??? warning "Do not run the installation of *Conda* and *DataLad* in the login node"
    
        HPC systems typically recommend using their login nodes only for tasks related to job submission, data management, and preparing jobscripts.
        Therefore, the execution of resource-intensive tasks such as *fMRIPrep* or building containers on login nodes can negatively impact the overall performance and responsiveness of the system for all users.
        Interactive sessions are a great alternative when available and **should** be used when creating the *DataLad* dataset.
        For example, in the case of systems operating SLURM, the following command would open a new interactive session:
        ```
        srun --nodes=1 --ntasks-per-node=1 --time=01:00:00 --pty bash -i
        ```

- [ ] Install *DataLad*.
    Generally, the most convenient and user-sandboxed installation (i.e., without requiring elevated permissions) can be achieved by using *Conda*, but other alternatives (such as *lmod*) can be equally valid:

    === "Install *DataLad* with *Conda*"
        - [ ] Get and install *Conda* if it is not already deployed in the system:
            ``` shell
            wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
            bash Miniconda3-latest-Linux-x86_64.sh
            ```
        - [ ] Install *DataLad*:
            ``` shell
            conda install -c conda-forge -y "datalad>=0.19" datalad-container
            ```

    === "Install *DataLad* in HPC with *lmod* enabled"
        - [ ] Check the availability and dependencies for a specific Python version (here we check 3.8.2):
            ``` bash
            module spider Python/3.8.2
            ```
        - [ ] Load Python (please note `ml` below is a shorthand for `module load`)
            ``` bash
            ml GCCcore/9.3.0 Python/3.8.2
            ```
        - [ ] Update *pip*:
            ``` bash
            python -m pip --user -U pip
            ```
        - [ ] Install *DataLad*:
            ``` bash
            python -m pip install --user "datalad>=0.19" datalad-container
            ```

- [ ] Check datalad is properly installed, for instance:
    ``` shell
    $ datalad --version
    datalad 0.19.2
    ```

    ??? bug "*DataLad* crashes (*Conda* installations)"

        *DataLad* may fail with the following error:
        ``` py
        ImportError: cannot import name 'getargspec' from 'inspect' (/home/users/cprovins/miniconda3/lib/python3.11/inspect.py)
        ```
        In such a scenario, create a *Conda* environment with a lower version of Python, and re-install datalad
        ``` shell
        conda create -n "datalad" python=3.10
        conda activate datalad
        conda install -c conda-forge datalad datalad-container
        ```

- [ ] Configure your Git identity settings.
    ``` shell
    cd ~
    git config --global --add user.name "Jane Doe"
    git config --global --add user.email doe@example.com
    ```

```

Reviewer: (I'd prefer you undo the changes on your branch rather than accepting this comment btw)
